src/System.Private.CoreLib/shared/System/AppDomain.cs

```diff
@@ -101,9 +101,9 @@ public static AppDomain CreateDomain(string friendlyName)
             throw new PlatformNotSupportedException(SR.PlatformNotSupported_AppDomains);
         }
 
-        public int ExecuteAssembly(string assemblyFile) => ExecuteAssembly(assemblyFile, null);
+        public int ExecuteAssembly(string assemblyFile) => ExecuteAssembly(assemblyFile, null!);
 
-        public int ExecuteAssembly(string assemblyFile, string?[]? args)
+        public int ExecuteAssembly(string assemblyFile, string?[] args)
```

Comment:
Why are we disallowing null here when it is a valid input? It does not make sense to me.

Reply:
Sorry it was my mistake. I wanted to do it in the one that takes params. I miss read.

Reply:
We agreed that we have a pattern in the framework that APIs that take a params shouldn't allow null.

Reply:
Now that I give it a second thought I don't think that is correct in this case. We're even calling our own overrides with null here. Will leave as nullable.